pip/index.py

```diff
@@ -291,7 +291,10 @@ def _validate_secure_origin(self, logger, location):
                 RemovedInPip7Warning,
             )
 
-    def find_requirement(self, req, upgrade):
+    def _get_indexes_locations(self, req):
+        """Returns locations found via self.index_urls
```

Comment:
I realize the rest of pip doesn't follow the convention, but should we start following PEP 257?